plugin/src/main/java/org/opensearch/ml/action/tasks/GetTaskTransportAction.java

```diff
@@ -84,6 +141,111 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLTask
             log.error("Failed to get ML task " + taskId, e);
             actionListener.onFailure(e);
         }
+    }
+
+    private void processRemoteBatchPrediction(MLTask mlTask, String taskId, ActionListener<MLTaskGetResponse> actionListener) {
+
+        Map<String, Object> transformJob = mlTask.getTransformJob();
 
+        Map<String, String> parameters = new HashMap<>();
+        for (Map.Entry<String, ?> entry : transformJob.entrySet()) {
+            if (entry.getValue() instanceof String) {
+                parameters.put(entry.getKey(), (String) entry.getValue());
+            } else {
+                log.debug("Value for key " + entry.getKey() + " is not a String");
+            }
+        }
+
+        if (parameters.containsKey("TransformJobArn") && parameters.get("TransformJobArn") != null) {
```

Comment:
Add some comments?
I think `TransformJobArn` and `TransformJobArn` only for Sagemaker.